pna_control/pna_control.py

```diff
@@ -114,7 +136,40 @@ def powersweep(startpower: float, endpower: float, numsweeps: int, centerf: floa
     file.write('POINTS: '+str(points)+'\n')
     file.close()
 
-    #run each sweep while increasing averages for each power
+    #run each sweep
     for i in sweeps:
-        getdata(centerf, span, temp, averages, i, edelay, ifband, points, outputfile)
+        get_data(centerf, span, temp, averages, i, edelay, ifband, points, outputfile)
+        # Iterate number of averages as power decreases
         averages = averages * ((10**(stepsize/10))**0.5)
+
+def name_datafile(outputfile,power,temp):
+    filename = outputfile+'_'+str(power)+'dB'+'_'+str(temp)+'mK'
+    filename = filename.replace('.','p')
+
+    return filename
+    
+def timestamp_folder(dir,meastype):
+    result = time.localtime(time.time())
+    output = str(result.tm_year)
+    if len(str(result.tm_mon)) < 2:
+        output = output + '0' + str(result.tm_mon)
+    else:
+        output = output + str(result.tm_mon)
+    if len(str(result.tm_mday)):
+        output = output + '0' + str(result.tm_mday) + '_' + str(result.tm_hour) + '_' + str(result.tm_min) + '_' + str(result.tm_sec)
+    else:
+        output = output + str(result.tm_mday) + '_' + str(result.tm_hour) + '_' + str(result.tm_min) + '_' + str(result.tm_sec)
+    
+    output = meastype+ '_' + output
```

Comment:
For things like `meastype` I suggest using an [enum](https://docs.python.org/3/library/enum.html) so that people can look up what allowed measurement types there are and it will fail if someone enters the wrong thing. These are more advanced features of the language but I think you will appreciate the guardrails.

Reply:
This would mean defining something like:

```python
from enum import Enum
class MeasType(Enum):
     FREQ_SCAN = 1
     TEMP_SWEEP = 2
     POWER_SWEEP = 3

def timestamp_folder(dir: str, meas: MeasType) -> str
...
    output = meas.name+ '_' + output
...
```

That way, the function gets called like: `timestamp_folder('foo', MeasType.FREQ_SCAN)` and if someone makes a mistake calling the function and types `timestamp_folder('foo', MeasType.FRQ_SCN)` they will get an error and the filenames will be much more uniform.

Reply:
I will add this as a separate patch. To do this, we need to actually decide all of the measurement types and have user files for each.